src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -1878,10 +1880,81 @@ protected override void ProcessRecord()
             }
         }
 
+        internal bool IsModuleInDenyList(string[] moduleDenyList, string moduleName, ModuleSpecification moduleSpec)
+        {
+            Debug.Assert(string.IsNullOrEmpty(moduleName) ^ (moduleSpec == null), "Either moduleName or moduleSpec can be specified");
+
+            var exactModuleName = string.Empty;
+            bool match = false;
+
+            if (!string.IsNullOrEmpty(moduleName))
+            {
+                // moduleName can be just a module name and it also can be a full path to psd1 from which we need to extract the module name
+                exactModuleName = Path.GetFileNameWithoutExtension(moduleName);
+            }
+            else if (moduleSpec != null)
+            {
+                exactModuleName = moduleSpec.Name;
+            }
+            
+            foreach (var deniedModuleName in moduleDenyList)
+            {
+                // use case-insensitive module name comparison
+                match = exactModuleName.Equals(deniedModuleName, StringComparison.InvariantCultureIgnoreCase);
+                if (match)
+                {
+                    string errorMessage = string.Format(CultureInfo.InvariantCulture, Modules.WinCompatModuleInDenyList, exactModuleName);
+                    InvalidOperationException exception = new InvalidOperationException(errorMessage);
+                    ErrorRecord er = new ErrorRecord(exception, "Modules_ModuleInWinCompatDenyList", ErrorCategory.ResourceUnavailable, exactModuleName);
+                    WriteError(er);
+                    break;
+                }
+            }
+            
+            return match;
+        }
+
+        internal List<T> FilterModuleCollection<T>(IEnumerable<T> moduleCollection)
```

Comment:
'private' accessibility for helper method?

Reply:
updated.